cmd/models/write.go

```diff
@@ -19,43 +19,61 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"os"
 
 	"github.com/openfga/cli/lib/cmd-utils"
 	"github.com/openfga/go-sdk/client"
 	"github.com/spf13/cobra"
 )
 
+func write(fgaClient client.SdkClient, text string) (string, error) {
+	body := &client.ClientWriteAuthorizationModelRequest{}
+
+	err := json.Unmarshal([]byte(text), &body)
+	if err != nil {
+		fmt.Printf("Failed to parse model due to %v", err)
+
+		return "", fmt.Errorf("failed to parse model due to %w", err)
+	}
+
+	model, err := fgaClient.WriteAuthorizationModel(context.Background()).Body(*body).Execute()
+	if err != nil {
+		fmt.Printf("Failed to write model due to %v", err)
+
+		return "", fmt.Errorf("failed to write model due to %w", err)
+	}
+
+	modelJSON, err := json.Marshal(model)
+	if err != nil {
+		fmt.Printf("Failed to write model due to %v", err)
+
+		return "", fmt.Errorf("failed to write model due to %w", err)
+	}
+
+	return string(modelJSON), nil
+}
+
 // writeCmd represents the write command.
 var writeCmd = &cobra.Command{
 	Use:   "write",
 	Short: "Write Authorization Model",
 	Args:  cobra.ExactArgs(1),
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		clientConfig := cmdutils.GetClientConfig(cmd)
+
 		fgaClient, err := clientConfig.GetFgaClient()
 		if err != nil {
 			fmt.Printf("Failed to initialize FGA Client due to %v", err)
-			os.Exit(1)
-		}
-		body := &client.ClientWriteAuthorizationModelRequest{}
-		err = json.Unmarshal([]byte(args[0]), &body)
-		if err != nil {
-			fmt.Printf("Failed to parse model due to %v", err)
-			os.Exit(1)
-		}
-		model, err := fgaClient.WriteAuthorizationModel(context.Background()).Body(*body).Execute()
-		if err != nil {
-			fmt.Printf("Failed to write model due to %v", err)
-			os.Exit(1)
+
```

Comment:
If we are returning the error, we should not be printing it (same elsewhere)

Reply:
removed